woocommerce/woocommerce
# [Experimental] Filter block: replace custom upgrade panel by Transforms API

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR:
- removes the upgrade and downgrade panel added in #43218.
- uses the Block Transforms API to add transform option to existing filter block.

Closes #43732  .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add the current Active Filters block to the page.
2. Customize that block:
  - Heading: move it below the filter block, change the content, customize the styling
  - Filter block: change default sidebar setting, customize styling.
3. Click on the icon of the filter wrapper block to open the block Transforms UI.
4. See Product Filter option there.
5. Hover on the `Product Filter`, see the preview that match with actual filter blocks.
6. Click on the `Product Filter`, see the existing filter block is replaced with new ones, with all customized setting carried over.
7. Repeat steps 2-6 with other filter blocks and see the same results.

<table>
<tr>
 <td> Before
 <td> After
<tr>
 <td><img src="https://github.com/woocommerce/woocommerce/assets/5423135/c0775eeb-9cc9-40b7-b5fd-99a291f52b5b" />
 <td><img src="https://github.com/woocommerce/woocommerce/assets/5423135/98c12f0c-d852-47a4-a8c3-8297b251ee73" />

</table>




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
